game-core/src/main/java/games/strategy/triplea/delegate/battle/TargetGroup.java

```diff
@@ -0,0 +1,94 @@
+package games.strategy.triplea.delegate.battle;
+
+import com.google.common.collect.Sets;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.Setter;
+import org.triplea.java.collections.CollectionUtils;
+
+/** Group of firing units and their targets. */
+@Getter
+@Setter
+public class TargetGroup {
+
+  private final Set<UnitType> firingUnitTypes;
+  private final Set<UnitType> targetUnitTypes;
+
+  public TargetGroup(final UnitType firingUnitType, final Set<UnitType> targetUnitTypes) {
+    firingUnitTypes = Sets.newHashSet(firingUnitType);
+    this.targetUnitTypes = targetUnitTypes;
+  }
+
+  public Collection<Unit> getFiringUnits(final Collection<Unit> units) {
+    return CollectionUtils.getMatches(units, Matches.unitIsOfTypes(firingUnitTypes));
+  }
+
+  public Collection<Unit> getTargetUnits(final Collection<Unit> units) {
+    return CollectionUtils.getMatches(units, Matches.unitIsOfTypes(targetUnitTypes));
+  }
+
+  public String getMessage(final GamePlayer player) {
```

Comment:
Nit, this method can be marked as static. It also seems to not really belong to this class as well (knowing it's static helps highlight that). Any thoughts to simply inline this to its single usage and avoid the indirection?

Reply:
This is a bit of a forward looking method as the message ideally would be more detailed to include which unit types are firing or being targeted.

Reply:
A TODO tagged with the forum discussion would have been ideal to have communicated that. Do you have an expectation when that would land (near-future, far-future?)

Reply:
Pushed it in line out of this class for now.